### PR TITLE
fix : Code block parser implementation  

### LINKED ISSUE #1834: [FR] supports markdown code format
### Description

supports markdown code format 

### Impact

coder

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by glunkad**
# Description
Code block parser implementation to support markdown code format

Fixes #1834 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by LucasXu0** [comment]
Hey, @glunkad. Thanks for contributing. Please add the test to cover your code. Here's an example, `share_markdown_test.dart`.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/parsers/code_block_node_parser.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+class CodeBlockNodeParser extends NodeParser {
+  const CodeBlockNodeParser();
+
+  @override
+  String get id => 'code_block';
+
+  @override
+  String transform(Node node) {
+    return '```$node.attributes[id]```';
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/parsers/code_block_node_parser.dart:11]: I think adding `\n` before and after the line should be better.

Comment (glunkad) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/parsers/code_block_node_parser.dart:11]: ok 👍🏻

**Comment by glunkad** [comment]
Hello @LucasXu0 , I am a bit lost and stuck can you guide me further.

**Comment by LucasXu0** [comment]
Sure, @glunkad. What's the question?

**Comment by glunkad** [comment]
Can you point me to the files which need to be changes in order to pass the tests?

**Comment by LucasXu0** [comment]
You don't need to modify anything else. The CI didn't pass because something goes wrong in Github Action.

**Pull request merged**